lib/matplotlib/animation.py

```diff
@@ -1156,8 +1160,9 @@ def _init_draw(self):
             self._draw_frame(next(self.new_frame_seq()))
         else:
             self._drawn_artists = self._init_func()
-            for a in self._drawn_artists:
-                a.set_animated(self._blit)
+            if iterable(self._drawn_artists):
```

Comment:
Good point. Keep in mind, when the animation module was first made, we
eschewed the animated property because it didn't really solve our issues.
That is why there is the disconnect there. However, it now makes sense to
move back towards it.
On Aug 27, 2015 11:08 PM, "Thomas A Caswell" <notifications@github.com>
wrote:

> In lib/matplotlib/animation.py
> <https://github.com/matplotlib/matplotlib/pull/4995#discussion_r38168184>:
>
> > @@ -1156,8 +1160,9 @@ def _init_draw(self):
> >              self._draw_frame(next(self.new_frame_seq()))
> >          else:
> >              self._drawn_artists = self._init_func()
> > -            for a in self._drawn_artists:
> > -                a.set_animated(self._blit)
> > +            if iterable(self._drawn_artists):
>
> If you are using blitting and return None blitting is effectively turned
> off (due to the logic it _post_draw) and if self._drawn_artists is not an
> iterable the code in_blit_draw` would raise.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4995/files#r38168184>.
>